src/highdicom/seg/sop.py

```diff
@@ -342,6 +365,55 @@ def __init__(
             **kwargs
         )
 
+        # General Reference
+        self.SourceImageSequence: List[Dataset] = []
+        referenced_series: Dict[str, List[Dataset]] = defaultdict(list)
+        for s_img in source_images:
+            ref = Dataset()
+            ref.ReferencedSOPClassUID = s_img.SOPClassUID
+            ref.ReferencedSOPInstanceUID = s_img.SOPInstanceUID
+            self.SourceImageSequence.append(ref)
+            referenced_series[s_img.SeriesInstanceUID].append(ref)
+
+        if len(referenced_series) > 1:
+            if is_multiframe and not is_tiled:
+                raise ValueError(
+                    'If source images are multiple-frame images that are '
+                    'not tiled, then only a single source image from a single '
+                    'series must be provided.'
+                )
```

Comment:
I really don't understand why there is different behaviour for tiled non-tiled. Can you please explain?

Reply:
My thinking goes as follows: If a multi-frame image is not tiled, then it likely represents an Enhanced CT/MR/PET Image and I wasn't sure how that should be handled. Would we need to check whether the number of frames in the image (each of the images if multiple should be allowed) match the first dimension of `pixel_array`?

Reply:
Addressed via a2c0d1d6342db23b20a3c3fab35056aceb8320f9

Reply:
For consistency, we should now also support `source_images: Sequence[Sequence[pydicom.Dataset]]` to allow users to specify multiple series of single-frame images. However, that will require further thought and work.